pybamm/spatial_methods/finite_volume.py

```diff
@@ -155,33 +120,33 @@ def gradient_matrix(self, domain):
         matrix = spdiags(data, diags, n - 1, n)
         return pybamm.Matrix(matrix)
 
-    def divergence(self, symbol, y_slices, boundary_conditions):
+    def divergence(self, symbol, discretised_symbol, boundary_conditions):
         """Matrix-vector multiplication to implement the divergence operator.
-        See :meth:`pybamm.BaseDiscretisation.gradient`
+        See :meth:`pybamm.SpatialMethod.gradient`
         """
         # Check that boundary condition keys are hashes (ids)
         for key in boundary_conditions.keys():
             assert isinstance(key, int), TypeError(
                 "boundary condition keys should be hashes, not {}".format(type(key))
             )
-        # Discretise symbol
-        discretised_symbol = self.process_symbol(symbol, y_slices, boundary_conditions)
         # Add Neumann boundary conditions if defined
         if symbol.id in boundary_conditions:
             # for the particles there will be a "negative particle" "left" and "right"
             # and also a "positive particle" left and right.
             lbc = boundary_conditions[symbol.id]["left"]
             rbc = boundary_conditions[symbol.id]["right"]
-            discretised_symbol = self.concatenate(lbc, discretised_symbol, rbc)
+            discretised_symbol = pybamm.NumpyModelConcatenation(
+                lbc, discretised_symbol, rbc
+            )
```

Comment:
This should be a `DomainConcatenation` (sorry, realise it's a mistake brought forward from previous implementation)